internal/dataplane/kongstate/route.go

```diff
@@ -255,17 +263,25 @@ func (r *Route) override(log logrus.FieldLogger, kongIngress *configurationv1.Ko
 	r.overrideByKongIngress(log, kongIngress)
 	r.overrideByAnnotation(log)
 	r.normalizeProtocols()
-	for _, val := range r.Protocols {
-		if *val == "grpc" || *val == "grpcs" {
-			// grpc(s) doesn't accept strip_path
-			r.StripPath = nil
-			break
+
+	if !r.ExpressionRoutes {
+		for _, val := range r.Protocols {
+			if *val == "grpc" || *val == "grpcs" {
+				// grpc(s) doesn't accept strip_path
+				r.StripPath = nil
+				break
+			}
 		}
 	}
```

Comment:
Could we move this logic into `normalizeProtocols`? That would save us at least one `if !r.ExpressionRoutes`. :)

Reply:
done